src/http/one/FieldParser.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_HTTP_ONE_FIELDPARSER_H
+#define SQUID_SRC_HTTP_ONE_FIELDPARSER_H
+
+#include "http/one/Parser.h"
+#include "HttpHeader.h"
+#include "parser/Tokenizer.h"
+
+namespace Http {
+namespace One {
+
+/** HTTP/1.x header field parser
+ *
+ * Works on a raw character I/O buffer and tokenizes the content into
+ * the field-lines of an HTTP/1 message:
+ *
+ * RFC 9112 section 5:
+ *  field-line   = field-name ":" OWS field-value OWS
+ */
+class FieldParser : public Http1::Parser
```

Comment:
Please do not derive a field parser from a message prefix parser. There is no "X is a Y" relationship between these two concepts.

Reply:
Sigh. You vetoed extracting the useful parts of `Http1::Parser` out into a separate class this one should have been importing them from. I see this is going to have to be even simpler for you.